### DISCUSSION TITLE
How to trigger DHW?

### DISCUSSION THREAD

**Discussion by Staacks** [Arnold-n/P1P2MQTT#126] [General]
I have been using P1P2MQTT for a while now and I am loving it. But so far I did not have much need to control DHW heating. Now I have gotten some solar panels and would like to use extra power for DHW: Reheating before sunset and increasing the setpoint when excess power is available.

Unfortunately, neither increasing the setpoint, nor enabling boost, nor toggling DHW mode, nor toggling heating would trigger water heating. The heatpump just remains idle until the temperature drops below the old setpoint minus hysteresis and does not care about anything I change.

To make it clear: I do not think that this is an P1P2MQTT issue (hence a discussion thread and not an issue). I can see the new settings on the heatpump's control panel as well as in the Daikin app and changing setpoint or boost mode in the Daikin app does not have any effect either. Also, I can see all the changed settings using ESPAltherma, so the settings find their way to the heatpump.

At the moment I am suspecting two things related to summer temperatures:
- I know that the heatpump cannot go beyond 45°C if ambient temperature is above 25°C (that is without electric backup heaters, which I have disabled). Since my normal setpoint is 45°C I could imagine that the heatpump will just internally use the upper limit due to ambient temperature, which is the same, so nothing happens. (Should do some testing at night, but at the moment the tank temperature will just hit the regular reheating point after sunset, which is why I am so interested in triggering it a bit earlier.)
- Right now the air temperature is reported as 24°C and still nothing happens. But I have also observed some situations in which the temperature would drop way below the reheating threshold (setpoint minus hysteresis). This often happens when the ambient temperature is high enough to cause the heatpump to enter "Fan only" mode. If that happens it seems to completely turn off the outdoor unit as none of its sensors would update at that point. But I have also seen that it would not reheat the water for a long time even if its temperature has dropped significantly. So far, this has not been bothering me, but I think that I remember reading somewhere that this is a firmware bug.

So, my question is: WTF, Daikin?

But more specifically to the P1P2MQTT users: Is anyone else struggling with this? Any ideas or workarounds? Or maybe I am just missing an important setting here?

(My heatpump is a Daikin Altherma 3 R W.)

**Comment by forTT78** [comment]
Hi Staacks,
I have the same situation. Could you check on your Daikin Altherma 3 R?

#9.3.6 [5-00]: Equilibrium: yes/no?
#9.3.7 [5-01]: Equilibrium temperatur?

Maybe it's looked because of these settings.

**Comment by Staacks** [comment]
Interesting, thanks, I did not know about these two settings. But these would prevent the backup heater from being used, which I do not want to use in any case¹. I have the problem that not even the heatpump will come on reliably to (re)heat the water. Also, aren't they just for heating, but not for DHW? At least that's how I read my manual.

¹The ratio of what I earn for selling the electric energy to what it costs to buy the electric energy is about the same as the ratio of using electric heating to the COP of the heatpump. So, for me it is not worth storing the energy with a COP of 1 if I can heat water on demand with a COP of 2 to 3. That of course might change with rising electricity costs.

**Reply by forTT78** [reply]
Hi,

I have a similar situation, at my heat pump I have the option "Warmhalten und Programm" [1] active. So, currently I have to modify the program at once to increase the temperature (eg. 55°C) to use the PV-power. 

In the situation you describe, how does the outside temperature measured by the outdoor unit relate to the outside temperature?
Same or with an offset? [2]

I read in the "Haustechnik-Dialog" forum that the heat pump can be switched to heating mode via the smart grid functionality ([SG1 / SG2](https://www.haustechnikdialog.de/Forum/p/3524643))
One active: recommendation to heat
Both active: force on (to heat)

I haven't implemented at my system yet, I'm still hoping for the feature via P1P2MQTT. :wink:



[1] I have only two possibilities: 
 "Warmhalten und Programm" or  "Warmhalten" - there is no option for "Programm only"
[2] [Modification temperature sensor](https://www.haustechnikdialog.de/Forum/p/3548695)

**Comment by Staacks** [comment]
Luckily, I do not have a problem with the outdoor temperature sensor. My outdoor unit is on the north side of our house and there is only a short period of 30 minutes before noon when it is hit by the sun directly through an narrow opening between the house and some trees. The rest of the day the temperature is spot on.

By now I also think that I am quite happy with my water heating. Not because I finally managed to control the heatpump properly, but because I virtually increased the size of our hot water tank. I noticed that the water sensor in the tank is quite in the middle and that we do not need that much headroom for our morning routine. So I placed a second temperature sensor a bit higher into the tank (through a connector external heating which we do not use) and use that to decide when to heat the water.

To do so, I have DHW turned off by default and let Home Assistant turn it on only when the new sensor gets cold enough. I also keep DHW disabled in the morning until my forecast for solar power is high enough. The result is that my wife and I can shower in the morning without ever having a problem with cold water so far and the system remains off despite a cold tank until there is enough solar power to heat the water (or until noon for extremely cloudy days). I also have a rule to enable DHW if the temperature has fallen too low to make it through the night to trigger DHW heating before the solar forecast gets too low in the evening.

So far it works perfectly and over the past few weeks 85% of the energy for our water came from solar power (without any batteries). The remaining 15% were necessary when the kids had a bath after sunset. (Things will probably get trickier when they are old enough to also shower in the morning.)

Therefore, while I am still interested in understanding how to control DHW heating more directly, it is not much of an issue anymore as most of the time the heatpump can just do it at some time in the morning when the delay has no consequence.

At the moment I am more wondering whether there is a good solution to do something similar for room heating. Here it seems like there is not way around requiring more power in the night than during the day. Reducing the heat output during the night just means paying it back before sunset to catch up in the morning... Maybe a battery will be a good addition to the system after all.

**Comment by martindell** [comment]
@Staacks thanks for starting this discussion - I have a related question that you maybe know the answer to?  

I’ve set my Daikin HP to use DHW in scheduled mode rather than reheat mode.  I have no solar, so I ‘super-heat’ the DHW tank overnight on cheap rate electricity.  All good, that works.

Ideally, I’d like to pull the DHW control & schedule into HomeAssistant but that’s where I get stuck. I’ve tried setting the Daikin scheduler to always on so I can control DHW ON/OFF via HomeAssistant - but without success.  I also can’t take advantage of reheat mode because of the temperature limit of that mode.

Did you discover anything about DHW control and scheduling by any chance?

**Comment by Staacks** [comment]
No, sorry, I am only using reheat mode. So, you turn on DHW via HA during the night and the heatpump ignores it entirely? That sounds worse than my struggle with reheat mode.

**Reply by kyerodev** [reply]
:) No worries.  So, I have DHW scheduled via the Diakin interface. The HP gets the DHW temp up to about 55 degrees, and then the booster heater kicks in to get it to 75.  

It's a really energy-inefficient method, I know - but my off-peak energy cost is 1/4 of my peak rate - so it gives my booster heater an effective 4X COP.  

This all works OK, but I'd prefer to have all the scheduling and control in HA (somehow). I've not figured out how to make that work.  I can enable DHW easily via HA, but DHW only switches ON when the internal Daikin schedule is triggered.  I can't seem to set up a Daikin DHW schedule that enables DHW as soon as HA enables DHW.

Not a biggie, it just seems such an obvious thing to want to do, I thought I must be missing something.

**Reply by Staacks** [reply]
You could try to do what I did and keep DHW off by default and in reheat with the smalles hysteresis you can set. Then turn on DHW from HA if the water gets too cold or turn on DHW and Boost if you want to use the cheap rate.

For me this works quite well and the heatpump starts heating the water immediately or up to 30 minutes after HA turned on DHW.

However, you need to be careful about your automations to leave DHW on until heating is done and to turn it off immediately afterwards to avoid frequent pointless reheating because of the small hysteresis. I solved this with a simple automation checks the water temperature every 5 minutes and turns on DHW if the water is too cold (and some other conditions that are not relevant here) and turns it off otherwise. This by itself would be a terrible idea as it would turn off DHW immediately after the temperature has reached the lower setpoint, but I added a condition to only execute this automation if the heatpump is not heating the water. This way, it will turn on DHW when the water is too cold, keep it on while the heatpump does not actually start heating (as the water remains too cold), keeps it on while the water is heating even if it is above the lower threshold (as the script is not executed while heating) and finally turns it off when the heatpump is done (as the water is warm AND the script is allowed to be executed again).

Of course, that condition cannot be the DHW setting, but something that actually indicated that the heatpump is heating the water. I use the three-way-valve position from my ESPAltherma (have been using this before adding the P1P2bridge), but "Valve_DHW_Tank" from the P1P2bridge seems to be the same.

Maybe this idea helps you. At least if the hysteresis prevents you from boosting in the night as the water is still too warm.

(BTW: Are you sure that heating to 75°C really saves enough energy when using the water later? I tried heating to a higher temperature during the day because of the free solar power, but removed that automation eventually for two reasons: a) My impression was that my familiy simply used more water as they mixed the same amount of hot water with additional cold water (that might depend on your shower head/controls). b) I realized that if the water at normal temperature lasts for 24h, I could just keep the water at that lower temperature with a better COP (I can sell excess solar power). IMHO, heating that high only helps if you would otherwise have to reheat at more expensive times (i.e. if your tank is rather small) and if you are disciplined enough to create the same water temperature in the shower by using less and mixing it with cold water.)

**Reply by kyerodev** [reply]
Thanks @Staacks I'll give the boost idea a try. I agree with you about the questionable wisdom of heating to 75 degrees.  This is a pretty typical DHW day for us (300 litre tank with the sensor near the top of the tank).  After it reaches temp at about 02:40, it loses 10 degrees by 09:00 without any usage at all.  The higher the tank temp, the higher the heat loss, I guess.

![Screenshot 2024-10-05 at 18 03 58](https://github.com/user-attachments/assets/5384c15e-ee39-4558-b254-0847be705017)

We do empty the tank quite frequently by the evening, so i am trying to avoid an expensive reheat during the day

**Reply by buennerbernd** [reply]
This is really a big loss. Do you have a hot water circulation installed? Is this always on? This would explain the big loos.
We have a Shelly Button installed that enables the circulation on demand and just for 2 minutes. That keeps the water hot.

**Reply by Staacks** [reply]
Good point. The circulation is not only a problem because of the loss from the hot pipes, but it can also thoroughly mix cold and hot water in your tank. I gained so much by disabling our circulation pump completely¹. The loss of comfort and the wasted water while waiting for hot water is rather small in my opinion.

¹Just keep in mind that a disabled circulation line can be a health risk because of the stale water. For this reason I set up a rule in HA to start the circulation pump for a few minutes whenever the heatpump starts heating water, simply because in that moment the mixing does not matter and the heat loss is minimal while preventing stale water in the circulation line.

Edit: Just to emphasize for anyone who might be confused by the terms: This is not about the circulating pump that the heatpump controls for room heating. This is about an external pump in a dedicated hot water circulation line and not controlled by P1P2MQTT.

**Comment by FlexxFR** [comment]
Very interesting discussion, I will start experimenting with a HA automation and just disable/enable DHW.

Although I do not understand the circulation pump part. How can you disable this and what's the purpose?
Is this the value: binary_sensor.hc_mode_circulation_pump ?
My impression was this is the pump needed for the floor heating.

**Reply by Staacks** [reply]
Sorry, the terms might be confusing:
The discussion about circulation was not about the circulation pump of the heatpump or the room heating, but about an external pump for a hot water circulation line (not even sure if this is the right term in English). This is a pipe that goes from the hot water tank through most parts of the house and then returns back to the tank and it includes a pump that circulates the hot water through it. The point is that this keeps the water in the circulation line hot and you do not have to wait so long for hot water as every faucet draws the hot water from the nearest point of the circulation line instead all the way through from the tank in the basement and through cold pipes. It is for convenience and you will save water with such a circulation line because you do not waste as much water while waiting for the hot water to arrive, but you will waste energy from heat losses and because it mixes the water in your tank (which probably was not an issue for gas heating where you just reheat whenever you want).

The point is: If your house does not have a circulation line, you can ignore that discussion :)

**Reply by FlexxFR** [reply]
ahaa..that makes sense. Wasn't even aware such systems existed :)

**Comment by gaumin** [comment]
Hi, an interesting discussion is evolving here. I also want to use excess solar power to heat my boiler. However, I want to go even further—not only triggering DWH mode but also forcing the use of the BUH (backup heater) to save the compressor resource of the heat pump.

The question is how to do it. The first thing that comes to mind is to play around with outdoor temperature readings. We already know that the BUH is used when the outdoor temperature is above 25°C. So, I’m wondering if I could fake the outdoor sensor readings (let's say set the outdoor temperature to 30°C) during the DWH cycle.

I’ve already noticed that this happens naturally during hot summer days. The only thing necessary is to find a way to fake the outdoor sensor readings.

Any ideas? Would this work?

**Reply by Timple** [reply]
My boiler has a hole for a heating element. These are rather cheap. Attach this to a relay module and you don't need anything from this project I guess..

**Reply by gaumin** [reply]
> My boiler has a hole for a heating element. These are rather cheap. Attach this to a relay module and you don't need anything from this project I guess..

@Timple, what kind of boiler do you use? This won't work with internal boilers from Daikin, I guess..

**Reply by Timple** [reply]
Ah, an external one... My bad

**Comment by Rewertius** [comment]
Hi everyone,
I’d like to jump in on this topic as well. I’m currently working on refining the control of my Daikin Altherma 3 HMT via P1P2. Heating mode is working fine so far – for example, I can adjust the flow temperature based on a schedule. The DHW entities are also available in Home Assistant and can be used in automations without any issues.

The problem: when I set the entity climate.hc_dhw_dhw_setpoint to heating mode, the command doesn’t seem to reach the heat pump or trigger any action. Home Assistant reports that heating mode is active, but on the device itself DHW stays off.

Has anyone run into this before or knows how to troubleshoot and fix it? Any help would be greatly appreciated!

**Reply by Rewertius** [reply]
This is the MQTT Message when i set climate.hc_dhw_dhw_setpoint to heating mode:
`* [MON] 2025-08-16 14:42:21 wr 0x35 0x3E to 0x01`
`* [MON] 2025-08-16 14:42:21 wr 0x35 0x40 to 0x01`

sensor.hc_dhw_dhw_setpoint = 50°C
sensor.hc_dhw_temperature_r5t_dhw_tank = 29°C
sensor.hc_sensors_temperature_outside = 20,8°C

**Reply by Rewertius** [reply]
Got it now: I had to set the device itself to “Keep Warm only” instead of “Program.” Try and error. No message from the controller or the heatpump, indicating that something is wrong. Another lesson learned.

**Comment by abbottcam** [comment]
Ive recently wired my ESPAltherma to trigger smart grid mode via 2 relays,  when home assistant see a surplus of PV power it triggers the heat pump to heat the DWH. I've also created a pulse meter to limit the heat pumps.
